tools/myst/host/exec.c

```diff
@@ -376,6 +378,31 @@ int exec_action(int argc, const char* argv[], const char* envp[])
 
     const char* rootfs = argv[2];
     const char* program = argv[3];
+    static const size_t extra_roothashes_size = 64;
+    char* extra_roothashes[max_roothashes];
+    ssize_t num_extra_roothashes = 0;
+
+    num_extra_roothashes = create_roothashes(
```

Comment:
We are not using roothash from the command line anymore, why don't we use the `roothashes` array directly?

Reply:
Why don't we make  `roothashes` an array of root hashes instead of file names containing the root hashes.

Reply:
We still use ``--roothash=<filename>`` when the app itself mounts the EXT2 fiel system (because the path to the image is not known by the ``myst`` command).

Reply:
We could make ``roothashes`` an array of hashes rather than filenames, but the archive packager expects files and files are still needed for the ``--roothash`` option.

Reply:
Can you elaborate when would a user want to mount an ext2 file system in their code? And what are the steps required for them to do so?

Reply:
Yes, that will be part of the ext2 doc. In short, a user can pass the path of the ext2 image to the app and then call ``mount(source, target, "ext2")``.

Reply:
I removed the creation of these roothash temporary files by reworking the implementation to use a new "roothashes" region (and .mystroothashes section).